src/main/kotlin/com/sourcegraph/cody/agent/protocol/util/Rfc3986UriEncoder.kt

```diff
@@ -0,0 +1,15 @@
+package com.sourcegraph.cody.agent.protocol.util
+
+object Rfc3986UriEncoder {
+
+  // todo solve this with library
+  fun encode(uri: String): String {
+    val isWindowsPath = uri.matches("^file:///[A-Z]:/.*".toRegex())
+    if (isWindowsPath) {
+      val found = "file:///([A-Z]):".toRegex().find(uri)!!
+      val partition = found.groups[1]!!.value
+      return uri.replace("file:///$partition:", "file:///${partition.lowercase()}%3A")
```

Comment:
basically, what we want to achieve is to replace the colon symbol (the one that appears after the partition name) with a `%3A`. is that right?

Reply:
Colon symbol and partition letter as lowercase for Windows paths - but possibly more.